src/nvim/getchar.c

```diff
@@ -1528,6 +1528,17 @@ int vgetc(void)
         c = utf_ptr2char(buf);
       }
 
+      // If mappings are enabled (i.e., not Ctrl-v) and the user directly typed
+      // something with a meta- or alt- modifier that was not mapped, interpret
+      // <M-x> as <Esc>x rather than as an unbound meta keypress.
```

Comment:
Would be good to keep the reference to the original discussion here.

Reply:
Good point, I'll add it.